src/components/structures/MatrixChat.js

```diff
@@ -923,6 +923,9 @@ export default React.createClass({
     },
 
     _viewHome: function() {
+        this.setStateForNewView({
```

Comment:
`_onLoggedIn` also sets this state and dispatches `view_home_page`, which should trigger this `_viewHome` when the action is processed:

https://github.com/matrix-org/matrix-react-sdk/blob/a3778dc822022e803c5b2a9d083f3889eeda4806/src/components/structures/MatrixChat.js#L1181-L1211

Should we remove the `setState` to change the view from ` _onLoggedIn`?

Reply:
At a glance, I think so. I'll do some testing and see what happens.